### DISCUSSION TITLE
Disable render-markdown in vim.lsp.buf.hover

### DISCUSSION THREAD

**Discussion by whoop-t** [MeanderingProgrammer/render-markdown.nvim#267] [Q&A] [answered]
render-markdown adds code background on vim.lsp.buf.hover(shift + k), is it possible to disable it effecting vim.lsp.buf.hover?
Verified that when I disable the plugin that the background no longer exists for the hover.

![Screenshot from 2024-12-20 23-27-01](https://github.com/user-attachments/assets/9eea64e6-0897-4354-aa85-d6a558c9e274)

**Comment by MeanderingProgrammer** [comment, marked as answer]
You can disable the plugin using a `buftype` override or you can change the background highlight associated with code blocks for hover docs.

To disable:

```lua
require('render-markdown').setup({
    overrides = {
        buftype = {
            nofile = { enabled = false },
        },
    },
})
```

To change code block background highlight:

```lua
require('render-markdown').setup({
    overrides = {
        buftype = {
            nofile = { code = { highlight = 'SOME_CLEAR_HIGHLIGHT_GROUP' } },
        },
    },
})
```

**Reply by whoop-t** [reply]
Perfect, tyty

**Comment by meijieru** [comment]
The `nofile` overrides disable it for codecompanion too. Is there a way to keep the codecompanion functionality enabled while disabling the hover one?

**Reply by MeanderingProgrammer** [reply]
If you know of a way to identify LSP hover docs other than the `nofile` filetype LMK and I can look into adding a feature to support it. Otherwise no, there is not a way to treat these 2 buffers differently.

**Reply by MeanderingProgrammer** [reply]
I added an ignore option here: https://github.com/MeanderingProgrammer/render-markdown.nvim/commit/05e6a6d119f90b99829ecb7eb85428a226c0c05f

If you figure out how to identify LSP hover docs you can disable this plugin on them via:

```lua
require('render-markdown').setup({
    ignore = function(buf)
        -- Insert logic to return true when buf is LSP hover doc
        -- Otherwise return false
    end,
})
```
